src/Features/Core/Portable/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,190 @@
+                new LocalizableResourceString(nameof(FeaturesResources.Make_field_readonly), WorkspacesResources.ResourceManager, typeof(WorkspacesResources)))
+        {
+        }
+
+        public override bool OpenFileOnly(Workspace workspace) => false;
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.ProjectAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                bool IsCandidateField(IFieldSymbol symbol) =>
+                        symbol.DeclaredAccessibility == Accessibility.Private &&
+                        !symbol.IsReadOnly &&
+                        !symbol.IsConst &&
+                        !symbol.IsImplicitlyDeclared &&
+                        symbol.Locations.Length == 1 &&
+                        !IsMutableValueType(symbol.Type);
+
+                // Stores the state for fields:
+                //  'isCandidate' : Indicates whether the field is a candidate to be made readonly based on it's declaration and options.
+                //  'written'     : Indicates if there are any writes to the field outside the constructor and field initializer.
+                var fieldStateMap = new ConcurrentDictionary<IFieldSymbol, (bool isCandidate, bool written)>();
+
+                void OnFieldWrite(IFieldSymbol field)
+                {
+                    Debug.Assert(IsCandidateField(field));
+                    Debug.Assert(fieldStateMap.ContainsKey(field));
+
+                    fieldStateMap[field] = (isCandidate: true, written: true);
+                }
+
+                (bool isCandidate, bool written) TryGetOrComputeIsCandidateField(IFieldSymbol fieldSymbol, AnalyzerOptions options, CancellationToken cancellationToken)
+                {
+                    return fieldStateMap.GetOrAdd(fieldSymbol, valueFactory: ComputeIsCandidate);
+
+                    (bool isCandidate, bool written) ComputeIsCandidate(IFieldSymbol field)
+                    {
+                        if (!IsCandidateField(field))
+                        {
+                            return (false, false);
+                        }
+
+                        var option = GetCodeStyleOption(field, options, cancellationToken);
+                        if (option == null || !option.Value)
+                        {
+                            return (false, false);
+                        }
+
+                        return (isCandidate: true, written: false);
+                    }
+                }
+
+                compilationStartContext.RegisterSymbolAction(symbolContext =>
+                {
+                    var _ = TryGetOrComputeIsCandidateField((IFieldSymbol)symbolContext.Symbol, symbolContext.Options, symbolContext.CancellationToken);
```

Comment:
why the variable?

Reply:
:grey_exclamation: use just `_` (no `var`) for discards